integration/dev_test.go

```diff
@@ -82,6 +84,66 @@ func TestDevNotification(t *testing.T) {
 	}
 }
 
+func TestDevGracefulCancel(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("graceful cancel doesn't work on windows")
+	}
+
+	tests := []struct {
+		name        string
+		dir         string
+		pods        []string
+		deployments []string
+	}{
+		{
+			name: "getting-started",
+			dir:  "examples/getting-started",
+			pods: []string{"getting-started"},
+		},
+		{
+			name:        "multi-config-microservices",
+			dir:         "examples/multi-config-microservices",
+			deployments: []string{"leeroy-app", "leeroy-web"},
+		},
+		{
+			name: "multiple deployers",
+			dir:  "testdata/deploy-multiple",
+			pods: []string{"deploy-kubectl", "deploy-kustomize"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ns, client := SetupNamespace(t)
+
+			procChan := make(chan *os.Process, 1)
+
+			go func() {
```

Comment:
I don't think I quite follow here. Immediately waiting for channel value in line 123 (`p := <-procChan`) makes this synchronous anyways, so isn't it equivalent to:
```diff
-			procChan := make(chan *os.Process, 1)
-			go func() {
-				proc, _ := skaffold.Dev().InDir(test.dir).InNs(ns.Name).RunWithProcess(t)
-				procChan <- proc
-			}()
-
-			p := <-procChan
+			p := skaffold.Dev().InDir(test.dir).InNs(ns.Name).RunWithProcess(t)
			client.WaitForPodsReady(test.pods...)
			client.WaitForDeploymentsToStabilize(test.deployments...)
...
```
also maybe rename `RunWithProcess` to `Start`?

Reply:
yeah thanks for pointing this out. originally when I wrote this test I was doing other things before blocking on the `<-procChan` call, but I removed those so there's no need to do that. updated

`Start` doesn't really capture that we're returning the `*os.Process` from the call, I'll rename to `StartWithProcess`